### DISCUSSION TITLE
Weird Connection Issues After DB Pasword Change

### DISCUSSION THREAD

**Discussion by darrylmorley** [prisma/prisma#9531] [Q&A] [answered]
Hey,

I have changed my db connection password. Updated my .env file with the new password in the connection string. regenerated prisma and it won't connect in dev. Yet db pull works.

```
Authentication failed against database server at `myserver`, the provided database credentials for `me` are not valid.
```
My .env:
```
DATABASE_URL="postgresql://myuser:mypassword@myserver:myport/mydb?schema=public"
```

`yarn prisma db pull`
```
darryl@Darryls-MBP xxx % yarn prisma db pull
yarn run v1.22.11
$ /Users/darryl/xxx/xxx/node_modules/.bin/prisma db pull
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "mydb", schema "public" at "myserver"

Introspecting based on datasource defined in prisma/schema.prisma …

✔ Introspected 5 models and wrote them into prisma/schema.prisma in 331ms
      
Run yarn prisma generate to generate Prisma Client.

✨  Done in 1.76s.
```

`yarn dev`
```
darryl@Darryls-MBP xxx % yarn dev
yarn run v1.22.11
$ next dev
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Loaded env from /Users/darryl/Projects/xxx/xxx/.env.local
info  - Loaded env from /Users/darryl/Projects/xxx/xxx/.env
info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
event - compiled successfully
error - PrismaClientInitializationError: 
Invalid `prisma.matrixProduct.findMany()` invocation:


  Authentication failed against database server at `myserver`, the provided database credentials for `me` are not valid.

Please make sure to provide valid database credentials for the database server at `myserver`.
```

How can it pull if it can't connect? 🤣

The weird thing, is that my production site is working fine with the new connection string.

Any ideas to get my dev setup working with the new password? Is there an issue with symbols in passwords or is the connection string maybe cached somewere?

**Comment by ryands17** [comment]
@darrylmorley 👋 

Could you check if you have permissions to connect from your system to the database? If it's working in production it means that something is preventing your current environment cannot connect to it due to some reason.

**Reply by darrylmorley** [reply]
@ryands17 Good point! I'll check if there is any kind of IP blocking going on 👍

**Reply by darrylmorley** [reply]
Nope, I can remotely login to the db from my terminal with my new password. This is just weird.

**Comment by darrylmorley** [comment, marked as answer]
I found the answer here https://github.com/prisma/prisma/issues/2177

The password needs to be encoded, although it is strange that this is working in prod, and db pull still works though..

**Reply by ryands17** [reply]
How come it's working on Production then? Special characters should be there as well.

**Reply by darrylmorley** [reply]
Beats me, I was just Googling that. I'm guessing that as the site is hosted with Vercel, they automatically encode env variables. How come you can introspect a database without an encoded password, but then the password needs to be encoded to perform any operations?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
